src/util/pac_utils.c

```diff
@@ -0,0 +1,150 @@
+/*
+    Authors:
+        Sumit Bose <sbose@redhat.com>
+
+    Copyright (C) 2022 Red Hat
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include <errno.h>
+#include <talloc.h>
+
+#include "config.h"
+
+#include "util/util.h"
+
+const char *check_check_pac_opt(const char *inp, uint32_t *check_pac_flags)
+{
+    int ret;
+    size_t c;
+    char **list = NULL;
+    uint32_t flags = 0;
+
+    if (strcasecmp(inp, CHECK_PAC_NO_CHECK_STR) == 0) {
+        flags = 0;
+        return inp;
+    }
+
+    ret = split_on_separator(NULL, inp, ',', true, true, &list, NULL);
+    if (ret != EOK) {
+        DEBUG(SSSDBG_OP_FAILURE, "Failed to split pac_check value.\n");
+        goto done;
+    }
+
+    for (c = 0; list[c] != NULL; c++) {
+        if (strcasecmp(list[c], CHECK_PAC_NO_CHECK_STR) == 0) {
+            DEBUG(SSSDBG_CONF_SETTINGS,
+                  "pac_check option [%s] can be only used alone.\n",
+                  CHECK_PAC_NO_CHECK_STR);
+            ret = EINVAL;
+            goto done;
+        } else if (strcasecmp(list[c], CHECK_PAC_PRESENT_STR) == 0) {
+            flags |= CHECK_PAC_PRESENT;
+        } else if (strcasecmp(list[c], CHECK_PAC_CHECK_UPN_STR) == 0) {
+            flags |= CHECK_PAC_CHECK_UPN;
+        } else if (strcasecmp(list[c], CHECK_PAC_UPN_DNS_INFO_PRESENT_STR) == 0) {
+            flags |= CHECK_PAC_UPN_DNS_INFO_PRESENT;
+            flags |= CHECK_PAC_CHECK_UPN;
+        } else if (strcasecmp(list[c], CHECK_PAC_CHECK_UPN_DNS_INFO_EX_STR) == 0) {
+            flags |= CHECK_PAC_CHECK_UPN_DNS_INFO_EX;
+        } else if (strcasecmp(list[c], CHECK_PAC_UPN_DNS_INFO_EX_PRESENT_STR) == 0) {
+            flags |= CHECK_PAC_UPN_DNS_INFO_EX_PRESENT;
+            flags |= CHECK_PAC_CHECK_UPN_DNS_INFO_EX;
+            flags |= CHECK_PAC_UPN_DNS_INFO_PRESENT;
+            flags |= CHECK_PAC_CHECK_UPN;
+        } else {
+            DEBUG(SSSDBG_OP_FAILURE, "Unknown value [%s] for pac_check.\n",
+                                     list[c]);
+            ret = EINVAL;
+            goto done;
+        }
+    }
+
+    if (check_pac_flags != NULL) {
+        *check_pac_flags = flags;
+    }
+
+    ret = EOK;
+done:
+    talloc_free(list);
+#ifndef HAVE_PAC_RESPONDER
+    if (ret == EOK && flags != 0) {
+        DEBUG(SSSDBG_CRIT_FAILURE,
```

Comment:
Perhaps `SSSDBG_IMPORTANT_INFO` fits better here as backtrace will be hardly useful.

Reply:
fixed